### PR TITLE
[BugFix] Fix missing ("next", "observation") key in dispatch of losses

### PR DESCRIPTION
**Pull request by Blonck**
## Description

Fix missing ("next", "dispatch") key of in_keys from IQL, DDPG, and A2C loss module.

## Types of changes

What types of changes does your code introduce? Remove all that do not apply:

- [x] Bug fix (non-breaking change which fixes an issue)

## Checklist

Go over all the following points, and put an `x` in all the boxes that apply.
If you are unsure about any of these, don't hesitate to ask. We are here to help!

- [x] I have read the [CONTRIBUTION](https://github.com/pytorch/rl/blob/main/CONTRIBUTING.md) guide (**required**)
- [x] My change requires a change to the documentation.
- [x] I have updated the tests accordingly (*required for a bug fix or a new feature*).

### CODE REVIEW

File: torchrl/objectives/a2c.py [hunk 1]
```diff
@@ -146,9 +147,10 @@ class A2CLoss(LossModule):
         ...     observation = torch.randn(*batch, n_obs),
         ...     action = spec.rand(batch),
         ...     next_done = torch.zeros(*batch, 1, dtype=torch.bool),
-        ...     next_reward = torch.randn(*batch, 1))
+        ...     next_reward = torch.randn(*batch, 1),
+        ...     next_observation = torch.randn(*batch, n_obs))
         >>> loss_val
-        (tensor(1.7593, grad_fn=<MeanBackward0>), tensor(0.2344, grad_fn=<MeanBackward0>), tensor(1.5480), tensor(-0.0155, grad_fn=<MulBackward0>))
+        (tensor(4.3483, grad_fn=<MeanBackward0>), tensor(1.4114, grad_fn=<MeanBackward0>), tensor(2.5165), tensor(-0.0252, grad_fn=<MulBackward0>))
```
Comment (vmoens) [inline torchrl/objectives/a2c.py:153]: maybe we don't want to print these values (such that we don't pollute PRs with diffs that aren't relevant?)

File: torchrl/objectives/iql.py [hunk 2]
```diff
@@ -156,9 +157,10 @@ class IQLLoss(LossModule):
         ...     observation=torch.randn(*batch, n_obs),
         ...     action=action,
         ...     next_done=torch.zeros(*batch, 1, dtype=torch.bool),
+        ...     next_observation=torch.zeros(*batch, n_obs),
         ...     next_reward=torch.randn(*batch, 1))
         >>> loss_val
-        (tensor(1.4535, grad_fn=<MeanBackward0>), tensor(0.8389, grad_fn=<MeanBackward0>), tensor(0.3406, grad_fn=<MeanBackward0>), tensor(3.3441))
+        (tensor(1.4535, grad_fn=<MeanBackward0>), tensor(0.7506, grad_fn=<MeanBackward0>), tensor(0.3406, grad_fn=<MeanBackward0>), tensor(3.3441))
```
Comment (vmoens) [inline torchrl/objectives/iql.py:163]: Ditto

**Review by vmoens — APPROVED** [review]
LGTM
see my minor comment

File: torchrl/objectives/a2c.py (hunk 1 above)
Comment (Blonck) [reply torchrl/objectives/a2c.py:153]: Good point, I will just remove it.

File: torchrl/objectives/a2c.py [hunk 3]
```diff
@@ -149,8 +149,6 @@ class A2CLoss(LossModule):
         ...     next_done = torch.zeros(*batch, 1, dtype=torch.bool),
         ...     next_reward = torch.randn(*batch, 1),
         ...     next_observation = torch.randn(*batch, n_obs))
-        >>> loss_val
-        (tensor(4.3483, grad_fn=<MeanBackward0>), tensor(1.4114, grad_fn=<MeanBackward0>), tensor(2.5165), tensor(-0.0252, grad_fn=<MulBackward0>))
```
Comment (vmoens) [inline torchrl/objectives/a2c.py]: We can still print something :) 
You could do
```
loss_actor, loss_val, *etc = fun_call(**kwargs)
loss_actor.backward() # for instance
```
anything that makes it feel like "this is a tensor"

**Review by vmoens — APPROVED** [review]
LGTM thanks!
